include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -157,6 +190,22 @@ namespace oneapi
 {
 namespace dpl
 {
+
+template <typename T>
+struct is_indirectly_device_accessible_iterator
+    : decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(std::declval<T>()))
+{
+    static_assert(
+        std::is_same_v<decltype(decltype(oneapi::dpl::__internal::__is_onedpl_indirectly_device_accessible_iterator(
+                           std::declval<T>()))::value),
+                       const bool>,
+        "Return type of is_onedpl_indirectly_device_accessible_iterator does not have the characteristics of a "
+        "bool_constant");
+};
+
+template <typename T>
+inline constexpr bool is_indirectly_device_accessible_iterator_v = is_indirectly_device_accessible_iterator<T>::value;
```

Comment:
...then
```suggestion
template <typename T>
inline constexpr bool is_indirectly_device_accessible_iterator_v = is_indirectly_device_accessible_iterator_t<T>::value;
```

Reply:
And the second point.
How we going to use this const in our code? Or in the user code?
Looks like it's just for our tests. If so, should we really want to have this const here?

Reply:
This is a part of the public API, provided for convenience as it common with this sort of `bool_constant` trait.  
We use the `bool_constant` in our code, not just our tests, but could just as easily use the value version of it now or in the future.  I think it is useful.

Reply:
I mean that I found only one usage of `is_indirectly_device_accessible_iterator_v` in our includes:
```C++
template <typename T>
inline constexpr bool is_indirectly_device_accessible_iterator_v = is_indirectly_device_accessible_iterator<T>::value;
```
It's ok?

Reply:
Yes.  We use `oneapi::dpl::is_indirectly_device_accessible<_Iter>` in the definition of `__is_passed_directly_device_ready`, but the value version of the trait is provide for convenience and flexibility of user code as part of the public API.

Reply:
Ok